File: src/Viriplaca.HR.Domain/WorkingTime.cs
namespace Viriplaca.HR.Domain;

public record WorkingTime
{
    public WorkingTime()
    {
    }

    public WorkingTime(decimal amount)
    {
        Amount = amount;
    }

    public const decimal HoursPerDay = 8;

    public decimal Amount { get; private init; }

    public int Days => (Amount / HoursPerDay).ToInt();

    public decimal Hours => (Amount % HoursPerDay).ToInt();

    public static WorkingTime FromDays(int days)
    {
        return new WorkingTime(days * HoursPerDay);
    }
}

File: src/Viriplaca.Enterprise.Web/src/models/leave/working-time.ts
import { humanizeDuration } from '@/plugins/dayjs';

export interface WorkingTime {
  amount: number;
  days: number;
  hours: number;
}

export const formatWorkingTime = (workingTime: WorkingTime): string => {
  if (workingTime.amount === 0) {
    return humanizeDuration(workingTime.days, 'days');
  }

  let result = '';
  if (workingTime.days > 0) {
    result += humanizeDuration(workingTime.days, 'days');
  }
  if (workingTime.hours > 0) {
    if (result) {
      result += ' ';
    }
    result += humanizeDuration(workingTime.hours, 'hours');
  }

  return result;
};
